lib/OpenQA/CLI/schedule.pm

```diff
@@ -0,0 +1,97 @@
+    my $params = $self->parse_params($args, $param_file);
+    my $tx = $client->start($client->build_tx(POST => $self->post_url, {}, form => $params));
+    my $res = $self->handle_result($tx, {pretty => 0, quiet => 1, links => 0, verbose => 0});
+    return $res if $res != 0;
+    my $json = $tx->res->json;
+    push @$job_ids, $json->{id} if defined $json->{id} && ref $json->{id} eq '';
+    push @$job_ids, @{$json->{ids}} if ref $json->{ids} eq 'ARRAY';
+    return 0 unless my $error = $json->{error};
+    print STDERR colored(['red'], $error, "\n");
+    return 1;
+}
+
+sub _monitor_jobs ($self, $client, $poll_interval, $job_ids, $job_results) {
+    while (@$job_results < @$job_ids) {
+        my $job_id = $job_ids->[@$job_results];
+        my $tx = $client->start($client->build_tx(GET => $self->url_for("experimental/jobs/$job_id/status"), {}));
+        my $res = $self->handle_result($tx, {pretty => 0, quiet => 1, links => 0, verbose => 0});
+        return $res if $res != 0;
+        my $job = $tx->res->json;
+        my $job_state = $job->{state} // NONE;
+        if (OpenQA::Jobs::Constants::meta_state($job_state) eq OpenQA::Jobs::Constants::FINAL) {
+            push @$job_results, $job->{result} // NONE;
+            next;
+        }
+        print "Job state of job ID $job_id: $job_state, waiting…\n";
+        sleep $poll_interval;
+    }
+}
+
+sub _compute_return_code ($self, $job_results) {
+    for my $job_result (@$job_results) {
+        return 2 unless OpenQA::Jobs::Constants::is_ok_result($job_result);
+    }
+    return 0;
+}
+
+sub command ($self, @args) {
+    die $self->usage
+      unless getopt \@args,
+      'param-file=s' => \my @param_file,
+      'm|monitor' => \my $monitor,
+      'i|poll-interval=i' => \my $poll_interval,
+      'r|set-return-code' => \my $set_return_code;
+    @args = $self->decode_args(@args);
+    my $client = $self->client($self->post_url);
+
+    my @job_ids;
+    my $create_res = $self->_create_jobs($client, \@args, \@param_file, \@job_ids);
+    return $create_res if $create_res != 0 || !($monitor || $set_return_code);
+
+    my @job_results;
+    my $monitor_res = $self->_monitor_jobs($client, $poll_interval // 10, \@job_ids, \@job_results);
+    return $monitor_res if $monitor_res != 0 || !$set_return_code;
+    return $self->_compute_return_code(\@job_results);
+}
+
+1;
+
+=encoding utf8
+
+=head1 SYNOPSIS
+
+  Usage: openqa-cli schedule [OPTIONS] DISTRI=… VERSION=… FLAVOR=… ARCH=… [ISO=… …]
+
+  Options:
+        --apibase <path>           API base, defaults to /api/v1
+        --apikey <key>             API key
+        --apisecret <secret>       API secret
+        --host <host>              Target host, defaults to http://localhost
+    -h, --help                     Show this summary of available options
+        --osd                      Set target host to http://openqa.suse.de
+        --o3                       Set target host to https://openqa.opensuse.org
```

Comment:
Do we really need to repeat all those parameters?

Reply:
I was asking myself the same question. At least it doesn't look like the Mojolicous command system would allows us to avoid it. I could of course ditch using it (and this weird way of making use of a code-comment that this way becomes more than just a comment after all) and just use a function (using a bunch of string literals) to print this. @kraih What's your take on this?

Reply:
Yes, there is currently no way to avoid it.

Reply:
k